### PR TITLE
implement edit defects

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

File: src/pages/testing/defects/defect-details/defect-details.spec.ts [hunk 1]
```diff
@@ -1,91 +1,91 @@
...
+//
+//   it('should check if the defect was added before', () => {
+//     comp.vehicleTest = navParams.get('test');
+//     comp.defect = navParams.get('defect');
+//     comp.vehicleTest.addDefect(addedDefect);
+//
+//     expect(comp.checkIfDefectWasAdded()).toBeTruthy();
+//   })
+//
+// });
```
Comment (mgherceanu) [inline src/pages/testing/defects/defect-details/defect-details.spec.ts]: please uncomment and resolve the tests for defect-details

File: src/pages/testing/test-creation/complete-test/complete-test.ts [hunk 2]
```diff
@@ -28,32 +37,33 @@ export class CompleteTestPage {
   }
 
   addDefect(): void {
-    this.navCtrl.push('AddDefectCategoryPage', {test: this.vehicleTest});
+    this.navCtrl.push('AddDefectCategoryPage', {
+      vehicleType: 'psv',
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/complete-test/complete-test.ts]: can you use the vehicle type from the vehicle object and not hardcode it?

File: src/pages/testing/test-creation/complete-test/complete-test.ts [hunk 3]
```diff
@@ -28,32 +37,33 @@ export class CompleteTestPage {
...
+      vehicleTest: this.vehicleTest,
+      defects: this.defectsCategories
+    });
   }
 
-  openDefect(defect: DefectModel): void {
-    return
+  openDefect(defect: DefectDetailsModel): void {
+    if (defect.deficiencyCategory != 'Advisory') {
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/complete-test/complete-test.ts]: can you please move the deficiency categories to an enum?

File: src/models/defects/defects.model.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import {AdditionalInfoModel} from "./additional-info.model";
+
+export interface DefectsModel {
+  category: DefectCategoryModel[]
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/models/defects/defects.model.ts]: done

File: src/pages/testing/defects/add-defect/add-defect.html [hunk 5]
```diff
@@ -12,21 +12,21 @@
...
       <span>
           <h3>
-            {{ item.ref }}
+            {{deficiency.ref}}
             <ion-badge item-end text-uppercase
-                       [color]="getBadgeColor(item.deficiencyCategory)"
-                       [ngStyle]="{'color':item.deficiencyCategory === 'Minor' ? 'black' : 'white' }">
-              {{ item.deficiencyCategory }}
+                       [color]="defectsService.getBadgeColor(deficiency.deficiencyCategory)"
+                       [ngStyle]="{'color':deficiency.deficiencyCategory === 'Minor' ? 'black' : 'white' }">
```
[the remark this answers is not in the document]
Comment (mgherceanu) [reply src/pages/testing/defects/add-defect/add-defect.html]: done

File: src/providers/defects/defects.service.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import {Injectable} from "@angular/core";
+
+@Injectable()
+export class DefectsService {
+
+  constructor() {}
+
+  getBadgeColor(category) {
+    switch (category.toLowerCase()) {
+      case 'dangerous':
```
Comment (adpetrescu) [inline src/providers/defects/defects.service.ts]: enum

File: src/providers/atf/atf.service.spec.ts [hunk 7]
```diff
@@ -8,7 +8,11 @@ describe('Provider: ATFService', () => {
   let atfService: AtfService;
   let storageService: StorageService;
   let spy: any;
-  let atfData = AtfDataMock.AtfData;
+
+  const atfData = AtfDataMock.AtfData;
+  let initialData: any[];
```
Comment (adpetrescu) [inline src/providers/atf/atf.service.spec.ts]: should this be any?

File: src/providers/atf/atf.service.ts [hunk 8]
```diff
@@ -7,6 +7,7 @@ import {STORAGE} from "../../app/app.enums";
 
 @Injectable()
 export class AtfService {
+  orderedArray: any[] = [];
```
Comment (adpetrescu) [inline src/providers/atf/atf.service.ts]: should this be any?

File: src/pages/testing/defects/defect-details/defect-details.html [hunk 9]
```diff
@@ -4,7 +4,10 @@
       Defect details
     </ion-title>
     <ion-buttons end>
-      <button ion-button block color="light" (click)="addDefect()" *ngIf="!checkIfDefectWasAdded()">Add defect</button>
+      <button ion-button block color="light" (click)="addDefect()">
+        <span *ngIf="!isEdit">Add Defect</span>
```
Comment (adpetrescu) [inline src/pages/testing/defects/defect-details/defect-details.html]: maybe use a ternary?

File: src/pages/testing/defects/advisory-details/advisory-details.html [hunk 10]
```diff
@@ -8,7 +8,8 @@
     <ion-title>Advisory details</ion-title>
     <ion-buttons end>
       <button ion-button (click)="submitAdvisory()">
-        Add Note
+        <span *ngIf="!isEdit">Add Note</span>
```
Comment (adpetrescu) [inline src/pages/testing/defects/advisory-details/advisory-details.html]: ternary

File: src/providers/atf/atf.service.spec.ts (hunk 7 above)
Comment (mgherceanu) [reply src/providers/atf/atf.service.spec.ts]: Done, initialData: AtfModel[]

File: src/providers/atf/atf.service.ts (hunk 8 above)
Comment (mgherceanu) [reply src/providers/atf/atf.service.ts]: Done, orderedArray: AtfModel[]

File: src/providers/defects/defects.service.ts (hunk 6 above)
Comment (mgherceanu) [reply src/providers/defects/defects.service.ts]: done

File: src/pages/testing/test-creation/complete-test/complete-test.ts (hunk 3 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/complete-test/complete-test.ts]: done

File: src/pages/testing/test-creation/complete-test/complete-test.ts (hunk 2 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/complete-test/complete-test.ts]: done

File: src/pages/testing/defects/advisory-details/advisory-details.html (hunk 10 above)
Comment (mgherceanu) [reply src/pages/testing/defects/advisory-details/advisory-details.html]: done

File: src/pages/testing/defects/defect-details/defect-details.html (hunk 9 above)
Comment (mgherceanu) [reply src/pages/testing/defects/defect-details/defect-details.html]: done

File: src/providers/atf/atf.service.ts [hunk 11]
```diff
@@ -15,4 +16,75 @@ export class AtfService {
     return from(this.storageService.read(STORAGE.ATFS))
   }
 
+  sortAndSearchATF(items: any[], filter: string, properties: string[]): any[] {
```
Comment (adpetrescu) [inline src/providers/atf/atf.service.ts]: do we need all those arrays to be any?

Comment (mgherceanu) [reply src/providers/atf/atf.service.ts]: we are still using the getters and the setters from the ATF model on the ATF details screen, we will refactor this when we will implement that screen (story)

**Pull request merged**